src/gui/transferlistsortmodel.cpp

```diff
@@ -40,16 +40,18 @@
 namespace
 {
     template <typename T>
-    bool customLessThan(const T left, const T right)
+    int customCompare(const T left, const T right)
     {
         static_assert(std::is_arithmetic_v<T>);
 
         const bool isLeftValid = (left >= 0);
         const bool isRightValid = (right >= 0);
 
         if (isLeftValid && isRightValid)
-            return left < right;
-        return isLeftValid;
+            return left - right;
+        if (!isLeftValid && !isRightValid)
+            return 0;
```

Comment:
This is not the common sense of a compare result and I strongly disagree using `0` to represent such case, this is too unexpected. This applies to other similar code.